coderd/autobuild/lifecycle_executor.go

```diff
@@ -140,42 +140,45 @@ func (e *Executor) runOnce(t time.Time) Stats {
 
 		eg.Go(func() error {
 			var job *database.ProvisionerJob
+			var auditLog *auditParams
 			err := e.db.InTx(func(tx database.Store) error {
 				// Re-check eligibility since the first check was outside the
 				// transaction and the workspace settings may have changed.
 				ws, err := tx.GetWorkspaceByID(e.ctx, wsID)
 				if err != nil {
-					log.Error(e.ctx, "get workspace autostart failed", slog.Error(err))
-					return nil
+					return xerrors.Errorf("get workspace by id: %w", err)
```

Comment:
review: IIRC the reason this was written this way was to avoid a single error causing an entire iteration to fail. Did the previous logic negatively impact the issue here? If not, I'd advise keeping the scope of the change small.

Reply:
This only returns errors for the transaction not for the `errgroup`. Seems like we were comitting the transaction regardless of whether we encountered an error or not which is bizarre. That shouldn't cause an entire evaluation iteration to abort early.

Reply:
Ah gotcha, I think I was operating on some stale cached assumptions about this code that are no longer true.